src/dune_engine/cached_digest.ml

```diff
@@ -256,7 +256,8 @@ let refresh_and_remove_write_permissions ~allow_dirs path =
             Path.Permissions.remove Path.Permissions.write stats.st_perm
           in
           Path.chmod ~mode:perm path;
-          refresh ~allow_dirs { stats with st_perm = perm } path
+          (* we know it's a file, so we don't allow directories for "safety" *)
```

Comment:
Not sure why "safety" is in quotes :-)

Reply:
Well, it's more for readability than safety. I'll remove the quotes though.